geode-for-redis/src/main/java/org/apache/geode/redis/internal/data/RedisString.java

```diff
@@ -469,7 +469,7 @@ public void handleSetExpiration(SetOptions options) {
 
   ////// methods that modify the "value" field ////////////
 
-  protected void valueAppend(byte[] bytes) {
+  protected synchronized void valueAppend(byte[] bytes) {
```

Comment:
Could you also update the comment on line 365 to not refer to "appendSequence" which no longer exists?